src/corehost/cli/exe/apphost/CMakeLists.txt

```diff
@@ -3,6 +3,12 @@
 
 cmake_minimum_required (VERSION 2.6)
 set(DOTNET_HOST_EXE_NAME "apphost")
+
+if (NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
```

Comment:
Yes, I have found that it cannot work on OSX. When loading libraries using dlopen, the rpath is ignored.